client/data/home/use-skip-current-view-mutation.ts

```diff
@@ -31,14 +32,31 @@ function useSkipCurrentViewMutation( siteId: number ): Result {
 					path: `/sites/${ siteId }/home/layout/skip`,
 					apiNamespace: 'wpcom/v2',
 				},
-				{ dev: query.dev },
+				{ query },
 				{
 					view: ( data as any ).view_name,
+					card,
 					...( reminder && { reminder } ),
 				}
 			);
 		},
 		{
+			onMutate( { card } ) {
+				const cachedData: Record< string, unknown > | undefined = queryClient.getQueryData(
+					getCacheKey( siteId )
+				);
+				const optimisticUpdate = {
+					...cachedData,
+					primary: cachedData.primary.filter( ( c ) => c !== card ),
+				};
+
+				queryClient.setQueryData( getCacheKey( siteId ), optimisticUpdate );
+			},
+			onError(/* err, _data, cachedData */) {
+				// We can revert the change if there's an error here, but do
+				// we actually want to?
+				// queryClient.setQueryData( getCacheKey( siteId ), cachedData );
+			},
 			onSuccess( data ) {
 				queryClient.setQueryData( getCacheKey( siteId ), data );
```

Comment:
I think we should remove this and rely on onMutate.

The order of cards returned by the skip endpoint is freshly randomized and different from what the onMutate function returns.

This leads to seeing a flash of a different card on a slow network connection.

When getting tracks working, I was also logging impresions of a card that was theoretically shown for a few ms.

Reply:
We need this to work for the single card case, though.